core/embed/extmod/modtrezorui/display.c

```diff
@@ -366,10 +371,10 @@ void display_icon(int x, int y, int w, int h, const void *data,
   int x0 = 0, y0 = 0, x1 = 0, y1 = 0;
   clamp_coords(x, y, w, h, &x0, &y0, &x1, &y1);
   display_set_window(x0, y0, x1, y1);
-  x0 -= x;
-  x1 -= x;
-  y0 -= y;
-  y1 -= y;
+  x0 -= MAX(x, 0);
+  x1 -= MAX(x, 0);
+  y0 -= MAX(y, 0);
+  y1 -= MAX(y, 0);
```

Comment:
Why this change?

Reply:
I did it to achieve starting position (0,0) when negative coordinates were passed for some reason, but thinking about it now, there might be legitimate use cases for this -> displaying only certain part of the icon, so i will drop this.

Reply:
> displaying only certain part of the icon, so i will drop this.

I had exactly this on my mind. The change disallows you to show just a part of the icon.

Reply:
fixed 5c3c68610ab8813c49a5126aabd4c63d26adbf04